aten/src/ATen/native/cpu/UpSampleMoreKernel.cpp

```diff
@@ -563,6 +422,280 @@ void upsample_trilinear3d_backward_kernel_impl(
   });
 }
 
+// Helper structs and methods for ti_cpu_upsample_linear
+
+// Linear Interpolation structure to compute output value in n-dimensional case.
+// - recursively compute interpolated output for each dimension
```

Comment:
Can we say that the interpolation methods that we use are separable, and as such we can compute the interpolation independently per dimension in a recursive way?
Maybe refer to #10482 for more context